src/metadata/metadata_status.h

```diff
@@ -217,32 +218,37 @@ static inline bool metadata_clear_dirty_sec_changed(
 			false);
 
 	if (was_dirty) {
-		is_dirty = cache->metadata.iface.clear_dirty(cache, line,
+		*became_clean = !cache->metadata.iface.clear_dirty(cache, line,
 				start, stop);
 	}
 
 	OCF_METADATA_BITS_UNLOCK_WR();
 
-	return was_dirty && !is_dirty;
+	sec_changed = was_dirty;
+
+	return sec_changed;
```

Comment:
Do we really need this additional variable?

Reply:
fixed